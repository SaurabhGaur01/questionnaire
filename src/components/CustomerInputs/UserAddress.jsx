import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setAddress } from '../../ducks/userInput/inputAddress';

const UserAddress = ({
    actionSetAddress,
    address,
}) =>{
    const addAddress = event => {
        actionSetAddress(event.target.value);
    }

    return (
        <div className="card card-container">
            <div className="card-body">
                <h5 className="card-title"> What's your Address ?</h5>
                <div className="form-group">
                    <input 
                        id="UserAddress-required" 
                        className="form-control" 
                        type="text" 
                        placeholder="User Address" 
                        onChange={addAddress}
                        value={address}
                    />
                </div>
            </div>
        </div>
    );
}    

UserAddress.propTypes = {
    actionSetAddress: PropTypes.func.isRequired,
    address: PropTypes.string.isRequired,
}

const mapDispatchAsProps = {
    actionSetAddress: setAddress,
}

export const mapStateToProps = state => ({
    address: state.userInput.address,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { UserAddress as TestableUserAddress };

export default hocChain(UserAddress);