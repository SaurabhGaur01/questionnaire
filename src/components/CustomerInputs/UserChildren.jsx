import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setChildren } from '../../ducks/userInput/inputChildren';
import { setChildrenPresent } from '../../ducks/userInput/hasChildren';

const UserChildren = ({
    actionSetChildren,
    numberOfChildren,
    actionSetChildrenPresent,
    hasChildren,
}) =>{
    const handleChange = (event) => {
        actionSetChildrenPresent(event.target.value);
    };

    const handleNoOfChildren = (event) => {
        actionSetChildren(parseInt(event.target.value));
    };

    return (
        <div className="card card-container">
            <div className="card-body">
                <h5 className="card-title">Do you have any children ?</h5>
                <div className="form-check">
                    <input 
                        className="form-check-input" 
                        type="radio" 
                        name="children" 
                        value="yes"
                        checked={hasChildren === "yes"}
                        onChange={handleChange}
                    />
                    <label className="form-check-label">
                        Yes
                    </label>
                </div>

                <div className="form-check">    
                    <input 
                        className="form-check-input" 
                        type="radio" 
                        name="children" 
                        value="no"
                        checked={hasChildren === "no"}
                        onChange={handleChange} 
                    />
                    <label className="form-check-label">
                        No
                    </label>
                </div>
                {
                    hasChildren === 'yes' && (
                        <div className="mt-5">
                            <h5 className="card-title"> How many children do you have ?</h5>
                            <div className="form-group">
                                <input 
                                    id="children-number" 
                                    className="form-control" 
                                    type="number" 
                                    placeholder="No of Children" 
                                    onChange={handleNoOfChildren}
                                    value={numberOfChildren}
                                />
                            </div>
                        </div> 
                    )
                }   
            </div>
        </div>
    );
}    

UserChildren.propTypes = {
    actionSetChildren: PropTypes.func.isRequired,
    numberOfChildren: PropTypes.number.isRequired,
    hasChildren: PropTypes.string.isRequired,
    actionSetChildrenPresent: PropTypes.func.isRequired,
}

const mapDispatchAsProps = {
    actionSetChildren: setChildren,
    actionSetChildrenPresent: setChildrenPresent,
}

export const mapStateToProps = state => ({
    numberOfChildren: state.userInput.numberOfChildren,
    hasChildren: state.hasChildren,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { UserChildren as TestableUserChildren };

export default hocChain(UserChildren);