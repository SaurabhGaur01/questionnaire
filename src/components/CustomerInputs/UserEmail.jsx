import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setEmail } from '../../ducks/userInput/inputEmail';

const UserEmail = ({
    actionSetEmail,
    email,
}) =>{
    const addEmail = event => {
        actionSetEmail(event.target.value);
    }

    return (
        <div className="card card-container">
            <div className="card-body">
                <h5 className="card-title"> What's your email ?</h5>
                <div className="form-group">
                    <input 
                        id="UserEmail-required" 
                        className="form-control" 
                        type="email" 
                        placeholder="User Email" 
                        onChange={addEmail}
                        value={email}
                    />
                </div>
            </div>
        </div>
    );
}    

UserEmail.propTypes = {
    actionSetEmail: PropTypes.func.isRequired,
    email: PropTypes.string.isRequired,
}

const mapDispatchAsProps = {
    actionSetEmail: setEmail,
}

export const mapStateToProps = state => ({
    email: state.userInput.email,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { UserEmail as TestableUserEmail };

export default hocChain(UserEmail);