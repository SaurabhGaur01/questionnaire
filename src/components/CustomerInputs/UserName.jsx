import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setName } from '../../ducks/userInput/inputName';

const UserName = ({
    actionSetName,
    firstName,
}) =>{
    const addName = event => {
        actionSetName(event.target.value);
    }

    return (
        <div className="card card-container">
            <div className="card-body">
                <h5 className="card-title"> What's your First Name ?</h5>
                <div className="form-group">
                    <input 
                        id="userName-required" 
                        className="form-control" 
                        type="text" 
                        placeholder="User Name" 
                        onChange={addName}
                        value={firstName}
                    />
                </div>
            </div>
        </div>
    );
}    

UserName.propTypes = {
    actionSetName: PropTypes.func.isRequired,
    firstName: PropTypes.string.isRequired,
}

const mapDispatchAsProps = {
    actionSetName: setName,
}

export const mapStateToProps = state => ({
    firstName: state.userInput.firstName,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { UserName as TestableUserName };

export default hocChain(UserName);