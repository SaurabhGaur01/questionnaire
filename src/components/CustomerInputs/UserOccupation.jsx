import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { setOccupation } from '../../ducks/userInput/inputOccupation';

const UserOccupation = ({
    actionSetOccupation,
    occupation,
}) =>{
    const handleChange = (event) => {
        actionSetOccupation(event.target.value);
    };

    return (
        <div className="card card-container">
            <div className="card-body">
                <h5 className="card-title">What's your Occupation ?</h5>
                <div className="form-check">
                    <input 
                        className="form-check-input" 
                        type="radio" 
                        name="occupation" 
                        value="EMPLOYED"
                        checked={occupation === "EMPLOYED"}
                        onChange={handleChange}
                        id="occupation-id"
                    />
                    <label className="form-check-label">
                        Employed
                    </label>
                </div>

                <div className="form-check"> 
                    <input 
                        className="form-check-input" 
                        type="radio" 
                        name="occupation" 
                        value="STUDENT" 
                        checked={occupation === "STUDENT"}
                        onChange={handleChange}
                    />
                    <label className="form-check-label">
                        Student
                    </label>
                </div>

                <div className="form-check">    
                    <input 
                        className="form-check-input" 
                        type="radio" 
                        name="occupation" 
                        value="SELF_EMPLOYED"
                        checked={occupation === "SELF_EMPLOYED"}
                        onChange={handleChange} 
                    />
                    <label className="form-check-label">
                        Self-employed
                    </label>
                </div>
            </div>
        </div>
    );
}    

UserOccupation.propTypes = {
    actionSetOccupation: PropTypes.func.isRequired,
    occupation: PropTypes.string.isRequired,
}

const mapDispatchAsProps = {
    actionSetOccupation: setOccupation,
}

export const mapStateToProps = state => ({
    occupation: state.userInput.occupation,
});

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps),
);

export { UserOccupation as TestableUserOccupation };

export default hocChain(UserOccupation);