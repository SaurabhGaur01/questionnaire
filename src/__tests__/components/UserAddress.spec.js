import React from 'react';
import { shallow } from 'enzyme';
import { TestableUserAddress, mapStateToProps} from '../../components/CustomerInputs/UserAddress';

describe('UserAddress Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            actionSetAddress: jest.fn(),
            address: 'mockAddress',
        };
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<TestableUserAddress {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    it('should call actionSetAddress when any input entered', () => {
        const renderedModule = shallow(<TestableUserAddress {...props} />);
        renderedModule.find('#UserAddress-required').at(0).simulate('change', {"target":{"value": 'mockKey'}});
        expect(props.actionSetAddress).toHaveBeenCalledTimes(1);
    });

    describe('mapStateToProps', () => {
        const addressMock = 'mockAddress';

        const mockedState = {
            userInput: {
                address: addressMock,
            }
        };

        it('should return the module state from the redux store', () => {
            const mockPropsFromRedux = mapStateToProps(mockedState);
            expect(mockPropsFromRedux).toEqual(
                {
                    address: addressMock, 
                }
            )
        })
    })
})