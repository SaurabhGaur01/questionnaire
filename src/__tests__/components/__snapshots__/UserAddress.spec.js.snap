// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAddress Component should render correctly 1`] = `
<div
  className="card card-container"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
       What's your Address ?
    </h5>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        id="UserAddress-required"
        onChange={[Function]}
        placeholder="User Address"
        type="text"
        value="mockAddress"
      />
    </div>
  </div>
</div>
`;
